Python/Tests/Analysis/AnalysisTest.cs

```diff
@@ -4301,6 +4317,34 @@ class C:
         }
 
         [TestMethod, Priority(0)]
+        public void AnnotatedAssign() {
+            var text = @"
+x : int = 42
+
+class C:
+    y : int = 42
+
+    def func(self):
+        self.abc : int = 42
+
+a = C()
+a.func()
+fob1 = a.abc
+fob2 = a.y
+fob3 = x
+";
```

Comment:
def f(val):
    print(val)

x: f(42)

val should be known to be an int
----
def f(val):
    print(val)

class C:
    y : f(42)

val should be known to be an int
---

def f(val):
    print(val)

def g():
    x : f(42)

val should have no type